### PR TITLE
Improve CLI API parity coverage

### PR DESCRIPTION
**Pull request by aronprins**
## Thinking Path

> - Paperclip is a control plane for AI-agent companies, with the CLI acting as a scriptable operator and agent interface to that control plane.
> - The REST API surface has grown across companies, agents, issues, routines, plugins, auth, workspaces, secrets, and operational inspection commands.
> - The CLI had drifted from that API surface: some commands were missing, some command shapes differed from docs/reference material, and several edge cases only failed during end-to-end local-source testing.
> - The local development runbook requires these tests to be disposable and isolated from a real `~/.paperclip`, `~/.codex`, or `~/.claude` installation.
> - This pull request adds broad CLI/API parity coverage, fixes the actionable bugs found during that pass, and records the reproducible test log under `doc/logs`.
> - The benefit is a more complete, scriptable CLI surface with regression coverage for the command families exercised by the parity run.

## What Changed

- Added or expanded CLI command coverage for access/auth, companies, agents, projects, goals, issues and subresources, routines, plugins, workspaces, activity/run/cost/dashboard inspection, assets, skills, secrets, tokens, prompt/wake flows, and local setup helpers.
- Fixed CLI/API parity bugs found during the run, including context profile patching, issue interaction optional payloads, malformed tree-hold errors, environment duplicate handling, configure invalid-section exit codes, worktree pnpm invocation, token agent ID resolution, plugin tool worker lookup, and routine webhook secret cleanup.
- Added missing CLI wrappers and route coverage for health/access, invite resolution URL forwarding, join status normalization, secret lifecycle commands, LLM docs routes, available-skill isolation, positive board-claim coverage, and interactive `connect` prompt-flow tests.
- Added a schema-backed `/api/openapi.json` route sufficient for CLI parity and `paperclipai openapi --json` smoke coverage.
- Added `doc/logs/2026-05-24-cli-api-parity-e2e-log.md` with the detailed living test/bug log and renamed the log directory from `doc/bugs` to `doc/logs`.
- Added `doc/plans/2026-05-23-cli-api-parity.md` and the OpenAPI parity reference used during the pass.

OpenAPI note: this PR intentionally does not try to subsume `feature/openapi-spec`. The OpenAPI implementation here is schema-backed and better than the earlier route-inventory stub, but `feature/openapi-spec` is the fuller/better OpenAPI branch because it includes exact mounted-route coverage tests and additional current route coverage. That branch should stay as its own PR and can supersede this OpenAPI route implementation.

## Verification

Targeted automated checks run:

- `pnpm exec vitest run server/src/__tests__/openapi-routes.test.ts`
- `pnpm exec vitest run server/src/__tests__/board-claim.test.ts`
- `pnpm exec vitest run cli/src/__tests__/connect.test.ts`
- `pnpm exec vitest run cli/src/__tests__/agent-lifecycle.test.ts`
- `pnpm exec vitest run server/src/__tests__/plugin-database.test.ts`
- `pnpm exec vitest run server/src/__tests__/routines-service.test.ts`
- `pnpm --dir cli typecheck`
- `pnpm --dir server typecheck`

Manual/local E2E verification:

- Ran the full disposable local-source CLI/API parity pass with isolated `PAPERCLIP_HOME`, `PAPERCLIP_CONFIG`, `PAPERCLIP_CONTEXT`, `PAPERCLIP_AUTH_STORE`, `CODEX_HOME`, and `CLAUDE_HOME` under `tmp/cli-api-parity`.
- Verified `DATABASE_URL` and `DATABASE_MIGRATION_URL` stayed unset for the scratch server.
- Verified live health and schema-backed OpenAPI responses on non-default port `3197`.
- Revoked created board/agent tokens and cleaned up temporary plugins, secrets, non-default environments, and project workspaces.
- See `doc/logs/2026-05-24-cli-api-parity-e2e-log.md` for the full command-by-command reproduction log.

Not run:

- Full `pnpm test`, `pnpm test:run`, or `pnpm build` were not run after the entire branch because the branch is broad and the parity pass used focused test/typecheck verification plus live isolated CLI reruns.

## Risks

- This is a broad PR and touches many CLI command modules, so review surface is high. The changes are grouped around one theme, but a split may be easier if maintainers prefer narrower PRs.
- The OpenAPI route in this PR is not the final/best OpenAPI implementation. `feature/openapi-spec` has stronger exact-route coverage and should remain the source for the dedicated OpenAPI PR.
- The living log is intentionally detailed and large. It is useful for reproducibility but adds documentation weight.
- No UI changes are intended; screenshots are not applicable.

> For core feature work, check [`ROADMAP.md`](ROADMAP.md) first and discuss it in `#dev` before opening the PR. Feature PRs that overlap with planned core work may need to be redirected — check the roadmap first. See `CONTRIBUTING.md`.

## Model Used

- OpenAI Codex, GPT-5-based coding agent in Codex desktop. Exact served model/context-window identifier was not exposed in the local app. Work used shell/Git/GitHub CLI tooling, local source inspection, targeted test execution, and live isolated Paperclip CLI/API smoke testing.

## Checklist

- [x] I have included a thinking path that traces from project context to this change
- [x] I have specified the model used (with version and capability details)
- [x] I have checked ROADMAP.md and confirmed this PR does not duplicate planned core work
- [x] I have run tests locally and they pass
- [x] I have added or updated tests where applicable
- [x] If this change affects the UI, I have included before/after screenshots
- [x] I have updated relevant documentation to reflect my changes
- [x] I have considered and documented any risks above
- [x] I will address all Greptile and reviewer comments before requesting merge

### CODE REVIEW

**Comment by aronprins** [comment]
@greptileai

File: server/src/routes/access.ts [hunk 1]
```diff
@@ -2610,6 +2639,90 @@ export function accessRoutes(
...
+      throw unauthorized("Board authentication required");
+    }
+    const keyId = (req.params.keyId as string).trim();
+    const key = await boardAuth.getBoardApiKeyForUser(keyId, req.actor.userId);
+    if (!key) throw notFound("Board API key not found");
+    const revoked = await boardAuth.revokeBoardApiKey(key.id);
+    if (!revoked) throw notFound("Board API key not found");
+
+    const companyIds = await boardAuth.resolveBoardActivityCompanyIds({
+      userId: req.actor.userId,
```
[the remark this answers is not in the document]
Comment (aronprins) [reply server/src/routes/access.ts:2710]: Fixed in 578c82d61: the delete route now validates `keyId` with `isUuidLike` before calling `getBoardApiKeyForUser`, returning a 400 for malformed IDs instead of letting the DB raise a UUID parse error.

File: server/src/services/routines.ts [hunk 2]
```diff
@@ -1880,6 +1880,9 @@ export function routineService(
         });
         return { deleted: true, revision: appended.revision };
       });
+      if (result.deleted && existing.secretId) {
+        await secretsSvc.remove(existing.secretId);
+      }
```
[the remark this answers is not in the document]
Comment (aronprins) [reply server/src/services/routines.ts:1892]: Fixed in 578c82d61: webhook secret removal after trigger deletion is now best-effort. Cleanup failures are caught and logged with routine/trigger/secret context so the already-successful delete response is preserved.

File: server/src/services/board-auth.ts [hunk 3]
```diff
@@ -160,6 +160,65 @@ export function boardAuthService(db: Db) {
...
+        name: boardApiKeys.name,
+        createdAt: boardApiKeys.createdAt,
+        lastUsedAt: boardApiKeys.lastUsedAt,
+        revokedAt: boardApiKeys.revokedAt,
+        expiresAt: boardApiKeys.expiresAt,
+      })
+      .from(boardApiKeys)
+      .where(eq(boardApiKeys.userId, userId))
+      .orderBy(sql`${boardApiKeys.createdAt} desc`);
+  }
```
[the remark this answers is not in the document]
Comment (aronprins) [reply server/src/services/board-auth.ts]: Fixed in 578c82d61: `listBoardApiKeys` now returns active keys by default, and the route supports `includeInactive=true` when callers need the full historical list.

File: cli/src/commands/client/token.ts [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+    const agent = await api.get<Agent>(apiPath`/api/agents/${trimmed}`);
+    if (!agent || agent.companyId !== companyId) throw new Error(`Agent not found: ${agentRef}`);
+    return agent;
+  }
+  const query = new URLSearchParams({ companyId });
+  const agent = await api.get<Agent>(`${apiPath`/api/agents/${trimmed}`}?${query.toString()}`);
+  if (!agent) throw new Error(`Agent not found: ${agentRef}`);
+  return agent;
+}
+
```
[the remark this answers is not in the document]
Comment (aronprins) [reply cli/src/commands/client/token.ts:230]: Fixed in 578c82d61: name-based agent resolution now checks `agent.companyId` against the requested company, matching the UUID lookup path.
